File: tdmu_movie_app_flutter_client/lib/models/review.dart
import 'auth_user.dart';

class Review {
  final int id;
  final int userId;
  final int movieId;
  final int rating;
  final String? comment;
  final DateTime createdAt;
  final AuthUser? user;

  const Review({
    required this.id,
    required this.userId,
    required this.movieId,
    required this.rating,
    this.comment,
    required this.createdAt,
    this.user,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: (json['id'] as num).toInt(),
      userId: (json['user_id'] as num).toInt(),
      movieId: (json['movie_id'] as num).toInt(),
      rating: (json['rating'] as num).toInt(),
      comment: json['comment'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
      user: json['user'] != null ? AuthUser.fromJson(json['user']) : null,
    );
  }
}

File: tdmu-movie-app-laravel-api/app/Models/Review.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Review extends Model
{
    protected $fillable = [
        'user_id',
        'movie_id',
        'rating',
        'comment',
    ];

    protected function casts(): array
    {
        return [
            'user_id' => 'integer',
            'movie_id' => 'integer',
            'rating' => 'integer',
            'created_at' => 'datetime',
            'updated_at' => 'datetime',
        ];
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}
